js/src/hooks/useAutoCreateAdsMCAccounts.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useCreateMCAccount from './useCreateMCAccount';
+import useUpsertAdsAccount from '.~/hooks/useUpsertAdsAccount';
+import useExistingGoogleAdsAccounts from '.~/hooks/useExistingGoogleAdsAccounts';
+import useExistingGoogleMCAccounts from '.~/hooks/useExistingGoogleMCAccounts';
+import useGoogleAdsAccount from './useGoogleAdsAccount';
+import useGoogleMCAccount from './useGoogleMCAccount';
+import { GOOGLE_MC_ACCOUNT_STATUS } from '.~/constants';
+
+/**
+ * Custom hook to handle the creation of Google Merchant Center (MC) and Google Ads accounts.
+ *
+ * @typedef {Object} AutoCreateAccountsStatus
+ * @property {boolean} accountsCreated Indicates if both the Google Ads and Google Merchant Center accounts have been successfully created.
+ * @property {boolean} accountCreationChecksResolved Indicates if the account creation checks (for existing accounts) have been resolved.
+ * @property {boolean} isCreatingAccounts Indicates if the Google Ads and/or Google Merchant Center account(s) are being created.
+ * @property {boolean} isCreatingOnlyAdsAccount Indicates if only the Google Ads account is currently being created.
+ * @property {boolean} isCreatingBothAccounts Indicates if both the Google Ads and Google Merchant Center accounts are currently being created.
+ * @property {boolean} isCreatingOnlyMCAccount Indicates if only the Google Merchant Center account is currently being created.
+ *
+ * @return {AutoCreateAccountsStatus} Object containing properties related to the account creation status.
+ */
+const useAutoCreateAdsMCAccounts = () => {
+	// Refs are used to avoid the re-render of the parent component.
+	const isCreatingBothAccountsRef = useRef( false );
+	const isCreatingAdsAccountRef = useRef( false );
+	const isCreatingMCAccountRef = useRef( false );
+	const initHasExistingMCAccountsRef = useRef( null );
+	const initHasExistingAdsAccountsRef = useRef( null );
+	const accountsCreatedRef = useRef( false );
+
+	const {
+		data: existingMCAccounts,
+		hasFinishedResolution: hasFinishedResolutionForExistingMCAccounts,
+	} = useExistingGoogleMCAccounts();
+
+	const {
+		existingAccounts: existingAdsAccounts,
+		hasFinishedResolution: hasFinishedResolutionForExistingAdsAccount,
+	} = useExistingGoogleAdsAccounts();
+
+	const {
+		hasFinishedResolution: hasFinishedResolutionForGoogleAdsAccount,
+		hasGoogleAdsConnection,
+	} = useGoogleAdsAccount();
+
+	const {
+		googleMCAccount,
+		hasFinishedResolution: hasFinishedResolutionForGoogleMCAccount,
+	} = useGoogleMCAccount();
+
+	const [ handleCreateAccount, { response } ] = useCreateMCAccount();
+	const [ upsertAdsAccount, { loading } ] = useUpsertAdsAccount();
+	const isGoogleMCConnected =
```

Comment:
@ankitrox This condition is different from what you have in `ConnectedGoogleComboAccountCard`. Aren't they supposed to be the same thing?